File: app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeekWeatherViewModelFactory.java
package com.example.perfectweatherallyear.ui.weekWeather;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.example.perfectweatherallyear.repository.WeatherRepositoryImp;

public class WeekWeatherViewModelFactory implements ViewModelProvider.Factory {
    private final WeatherRepositoryImp repository;

    public WeekWeatherViewModelFactory(WeatherRepositoryImp repository) {
        this.repository = repository;
    }

    @Override
    public <T extends ViewModel> T create(Class<T> modelClass) {
        return (T) new WeekWeatherViewModel(repository);
    }
}

File: app/src/main/java/com/example/perfectweatherallyear/ui/weekWeather/WeekWeatherViewModelFactory.kt
package com.example.perfectweatherallyear.ui.weekWeather

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.example.perfectweatherallyear.repository.WeatherRepository
import javax.inject.Inject

class WeekWeatherViewModelFactory @Inject constructor(val repository: WeatherRepository) : ViewModelProvider.Factory {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(WeekWeatherViewModel::class.java))
            return WeekWeatherViewModel(repository) as T
        else
            throw IllegalArgumentException("Unable to construct viewModel")
    }
}